### ISSUE TITLE
[FEATURE] Execute commands in parallel

### ISSUE THREAD

**Issue by lechnerc77** [SAP/terraform-provider-btp#9]
The Terraform provider should be capable of a parallel execution of commands that create/update/delete resources in parallel.

As the Terraform provider is delegating this to the BTP CLI backend, this capability must be first implemented in the BTP CLI backend to use it in the provider

**Comment by vipinvkmenon** [comment]
By default, independent resources are created [parallely](https://developer.hashicorp.com/terraform/language/resources/behavior#resource-dependencies). This means from our provider's perspective we don't really need to do much for the implementation.

Another aspect that was verified is how the provider handles authentication. The provider makes use of sessions for each step. Since the authentication happens in the [Configure](https://developer.hashicorp.com/terraform/plugin/framework/internals/rpcs#configureprovider-rpc), the authentication is done whenever `Configure` function is called. `Configure` is called many [commands](https://developer.hashicorp.com/terraform/plugin/framework/internals/rpcs#rpcs-and-framework-functionality) like `plan`, `apply` & `delete`. 


For E.g: Since `terraform apply` also explicitly runs `terraform plan`, the provider authenticates twice.

TLDR: We can close this backlog as there is nothing more that needs to be done from implementation perspective.
